contribs/drt/src/main/java/org/matsim/contrib/drt/run/DrtConfigGroup.java

```diff
@@ -646,4 +648,13 @@ public DrtInsertionSearchParams getDrtInsertionSearchParams() {
 	public Optional<DrtRequestInsertionRetryParams> getDrtRequestInsertionRetryParams() {
 		return Optional.ofNullable(drtRequestInsertionRetryParams);
 	}
+
+	/**
+	 * Convenience method that brings syntax closer to syntax in, e.g., {@link PlansCalcRouteConfigGroup} or {@link PlanCalcScoreConfigGroup}
+	 */
+	public final void addDrtInsertionSearchParams(final DrtInsertionSearchParams pars) {
+		testForLocked() ;
```

Comment:
Maybe `testForLocked()` could be inlined into `addParameterSet()` (at least in `ReflectiveConfigGroupWithConfigurableParameterSets` if not `ReflectiveConfigGroup`) ?

Reply:
I moved it up one level.  I would be fine with moving it up more than that, but don't have time to check the fallout.  In general, this is a device that I introduced so that certain params that cannot change after the scenario was generated can be locked.  But it is not used systematically beyond that.